### PR TITLE
AFE form Updates

### PR DESCRIPTION
**Pull request by hannahbergam**
This PR does two things:

1. Edits the string about supporting 'you and your students'
2. Removes the form additions requesting address if a user requests the 'Thank you kit'. See the screenshots below to illustrate the change. Additionally, this required altering some of the save functionality. I opted not to change the afe submission that exists in afe_enrollment.rb. Because the address was already optional, this should work as-is.

Note: I reached out regarding this form being translated. It's not currently top priority, but I've created a followup task to translate all the strings on this page here: https://codedotorg.atlassian.net/browse/ACQ-1199

Before:
Screenshot 2023-11-20 at 2 25 48 PM

After:
Screenshot 2023-11-20 at 2 23 24 PM

## Warning!!

We have entered Pixel Lock for Hour of Code!

Computer Science Education Week will be happening from Dec 4 - Dec 10. Alongside this event, we will
be launching our new Hour of Code activity. Please consider any risk introduced in this PR that
could affect Dance Lab, instructions, saving and logging student progress, caching, or anything
related to the Hour of Code activities, old or new. Even small changes, such as a different button
color, are considered significant during this time. If this change will affect the new Hour of Code
activity in any way, join the morning change review to get your changes approved prior to merging.
Reach out to the Student Labs team for more details!

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-614

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx [hunk 1]
```diff
@@ -440,76 +414,3 @@ const styles = {
...
-  return (
-    <div>
-      <div>
-        By checking the box above, I consent to Amazon sharing my email address
-        and school address with its third party vendor, Corporate Imaging
-        Concepts, LLC, solely in order to fulfill my request. I understand
-        Amazon's vendor will email me a promo code to allow me to select and
-        redeem my Thank You Kit items.
-      </div>
-      <FieldGroup
```
Comment (bethanyaconnor) [inline apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx:460]: Could be done as a follow-up but should these parameters also be removed from [the controller](https://github.com/code-dot-org/code-dot-org/blob/651b9e0431f76670108ef405b7192f8b49a7c2b8/dashboard/app/controllers/api/v1/amazon_future_engineer_controller.rb#L22)?

Comment (hannahbergam) [reply apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx:460]: Ahh yes I was grappling with that in the description lol trying to decide if it was worth the time to do with this ticket. I can just do it now😂

Comment (bethanyaconnor) [reply apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx:460]: oh I missed that sentence 🤦‍♀️ feel free to do it as a follow-up if you prefer!

**Review by bethanyaconnor — APPROVED** [review]

File: apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx (hunk 1 above)
Comment (hannahbergam) [reply apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx:460]: Made a ticket https://codedotorg.atlassian.net/browse/ACQ-1207

**Pull request merged**